### PR TITLE
Get attribute values for non-type restricted relationships eg storage

### PR DESCRIPTION
**Pull request by kehh**
* Get label for placements from the default UI for relationships
  - Bundles for screens now take the pattern of ca_tablename.bundle

### CODE REVIEW

**Comment by collectiveaccess** [comment]
This is going to break existing configurations. What is the motivation for this change?

**Comment by kehh** [comment]
tl;dr: Values on interstitial attributes don't save without this change

The change in the setter was because the request parameters that were coming across didn't have a `__` in them for attributes, only intrinsics, so I added an extra possible field name for the request parameter.

The code that was attempting to pull in labels for intrinsics from the first available UI wasn't pulling labels in because:

* The bundles are in the format `ca_tablename.bundlename` in the ca_editor_ui_placement, not simply `bundlename` so the call to getPlacementsForBundle would always return an empty array. I then added the enhancement to find a type specific UI if it exists rather than just the first UI for the relationship table using `ca_editor_uis::loadDefaultUI` rather than `ca_editor_uis::find`.
My config is:

```
history_tracking_policies = {
        defaults = {
                ca_objects = current_location
        },
        policies = {
                current_location = {
                        name = _(Location),
                        table = ca_objects,
                        mode = workflow, # movements or workflow
                        elements = {
                                ca_storage_locations = {
                                        __default__ = {
                                                date = ca_objects_x_storage_locations.effective_date,
                                                setInterstitialElementsOnAdd = [effective_date,currentLocationFitness,currentLocationNote,movementAuthoriser,movementAuthDate,movementContact,movementNote,movement_reason,plannedDate],
                                                useDatePicker = 1,
                                                useHierarchicalBrowser = 1,
                                                template ="
                                                <l>^ca_storage_locations.hierarchy.preferred_labels.name%delimiter=_➜_</l>
                                                <l>^ca_storage_locations.hierarchy.preferred_labels.name%delimiter=_➜_</l>
                                                <ifdef code='ca_objects_x_storage_locations.effective_date'>MOVEMENT DATE: ^ca_objects_x_storage_locations.effective_date</ifdef>
                                                <ifdef code='ca_storage_locations.locationType'>^ca_storage_locations.locationType</ifdef>
                                                <ifdef code='ca_objects_x_storage_locations.plannedDate'><br>PLANNED DATE: ^ca_objects_x_storage_locations.plannedDate</ifdef>
                                                <ifdef code='ca_objects_x_storage_locations.movementContact'><br>CONTACT: ^ca_objects_x_storage_locations.movementContact</ifdef>
                                                <ifdef code='ca_objects_x_storage_locations.movementAuthoriser'><br>AUTHORISED BY: ^ca_objects_x_storage_locations.movementAuthoriser</ifdef>
                                                <ifdef code='ca_objects_x_storage_locations.movementAuthDate'><br>AUTHORISED ON: ^ca_objects_x_storage_locations.movementAuthDate</ifdef>
                                                <ifdef code='ca_objects_x_storage_locations.movement_reason'><br>REASON: ^ca_objects_x_storage_locations.movement_reason</ifdef>
                                                <ifdef code='ca_objects_x_storage_locations.movementNote'><br>NOTE: ^ca_objects_x_storage_locations.movementNote</ifdef>
                                                ",
                                                trackingRelationshipType = contains,
                                                restrictToRelationshipTypes = [contains]
                                        }
                                }
                        }
                },
                identification = {
                        name = _(Identification),
                        table = ca_objects,
                        mode = workflow, # movements or workflow
                        elements = {
                                ca_occurrences = {
                                        scientificName = {
                                                date = ca_objects_x_occurrences.dateIdentified,
                                                useDatePicker = 1,
                                                restrictToRelationshipTypes = [describes],
                                                setInterstitialElementsOnAdd = [dateIdentified, verbatimDateIdentified, identifiedBy, identifierRole, typeStatus, typeOfType, typeStatusQualifier, identificationQualifier, identificationReferences, idVerificationStatus, identificationRemarks, typifiedName, verbatimIdentification],
                                                template =
                                                 "<table class='listtable identification ^ca_objects.type_id'>
                                                      <thead>
                                                      <tr>
                                                          <th>Name</th>
                                                          <th>Qualifier</th>
                                                          <th>Family</th>
                                                          <th>Date</th>
                                                          <th>Type Status</th>
                                                          <th>Identified By</th>
                                                          <th>Verification Status</th>
                                                          <th>Note</th>
                                                      </tr>
                                                      </thead>
                                                      <tbody>
                                                      <tr class='^ca_objects_x_occurrences.typeStatus ^ca_objects_x_occurrences.typeOfType'>
                                                          <td><l>^ca_occurrences.preferred_labels.name</l></td>
                                                          <td class='more'>^ca_objects_x_occurrences.identificationQualifier</td>
                                                          <td>^ca_occurrences.family</td>
                                                          <td>^ca_objects_x_occurrences.dateIdentified</td>
                                                          <td>
                                                            ^ca_objects_x_occurrences.typeStatus ^ca_objects_x_occurrences.typeOfType
                                                          </td>
                                                          <td>^ca_objects_x_occurrences.identifiedBy</td>
                                                          <td>^ca_objects_x_occurrences.idVerificationStatus</td>
                                                          <td class='more'>^ca_objects_x_occurrences.identificationRemarks</td>
                                                      </tr>
                                                      </tbody>
                                                  </table>"
                                        }
                                }
                        }
                }
        }
 }
```

Without this change the `identification` policy sets interstitial intrinsics and attributes since my previous PR.

With this change the `current_location` policy can set both  interstitial intrinsics and attributes since my previous PR.

**Comment by collectiveaccess** [comment]
Thanks for the explanation. I'll test this and merge soon.
